src/validation.cpp

```
@@ -6321,7 +6321,7 @@ Chainstate& ChainstateManager::ActivateExistingSnapshot(uint256 base_blockhash)
     LogInfo("[snapshot] switching active chainstate to %s", m_snapshot_chainstate->ToString());
 
     // Mempool is empty at this point because we're still in IBD.
-    Assert(m_active_chainstate->m_mempool->size() == 0);
+    Assert(!m_active_chainstate->m_mempool || m_active_chainstate->m_mempool->size() == 0);
```

Is this now weaker that the original assertion, because I think it will now silently accepts a null mempool pointer?

---

It is weaker, yes, but we similarly guard in other situations. We also still have the assertion in ActivateSnapshot to guard this condition at a more critical point in time.